### PR TITLE
Adds category resource to the Commercetools provider

### PR DESCRIPTION
**Pull request by Ardjan-Aalberts**
Adds category resource to the provider. Hierarchy and assets are supported.

Based on #145 by @viggin543

### CODE REVIEW

File: commercetools/resource_category.go [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+
+func resourceCategoryGetAssetSources(i map[string]interface{}) []commercetools.AssetSource {
+	var sources []commercetools.AssetSource
+	for _, item := range i["sources"].([]interface{}) {
+		s := item.(map[string]interface{})
+
+		source := commercetools.AssetSource{
+			URI:         s["uri"].(string),
+			Key:         s["key"].(string),
+			ContentType: s["content_type"].(string),
```
Comment (viggin543) [inline commercetools/resource_category.go:387]: content_type and key props  are optional and might be nil ( casting nil to string will cause panic ).

I see you have a test where you create a category with an asset that has no content type .
if it works thas weird.

File: commercetools/resource_category.go [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+
+	for _, raw := range input {
+		i := raw.(map[string]interface{})
+
+		name := commercetools.LocalizedString(expandStringMap(i["name"].(map[string]interface{})))
+		description := commercetools.LocalizedString(expandStringMap(i["description"].(map[string]interface{})))
+		sources := resourceCategoryGetAssetSources(i)
+
+		result = append(result, commercetools.AssetDraft{
+			Key:         i["key"].(string),
```
Comment (viggin543) [inline commercetools/resource_category.go:369]: key prop is optional

**Review by viggin543 — APPROVED** [review]
clean work

**Review by viggin543 — COMMENTED** [review]
you also don't support adding new assets ( only  update existing sources of an asset )

**Review by qdegraaf — CHANGES_REQUESTED** [review]
LGTM functionality wise!

Two changes I'd like to request is that we now have support for a new way of generating docs using the `Description` fields of the schema. If you could fill in these fields (see other resource for inspiration) this would save us having to fill them out down the line when we definitively move away from read the docs to the new set-up.

Second, in order to keep the tests a bit more consistent, master now has `CheckDestroy` funcs for all resources. If you could add one here as well that'd be grand.

**Review by qdegraaf — APPROVED** [review]

**Comment by Ardjan-Aalberts** [comment]
> you also don't support adding new assets ( only update existing sources of an asset )

Hi @viggin543, creating assets is supported. The assets are added to the draft after the draft is initially created. This because we first check if the Terraform config contains assets.

Thanks for reviewing the pull request!

**Pull request merged**